pallets/liquidity-pools/src/lib.rs

```diff
@@ -1013,6 +1013,44 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Initiate the recovery of assets which were sent to an incorrect
+		/// contract by the account represented by `domain_address`.
+		///
+		/// NOTE: Asset and contract addresses in 32 bytes in order to support
+		/// future non-EVM chains.
+		///
+		/// Origin: Root.
+		#[pallet::call_index(17)]
+		#[pallet::weight(T::WeightInfo::update_tranche_hook())]
+		pub fn recover_assets(
+			origin: OriginFor<T>,
+			domain_address: DomainAddress,
+			incorrect_contract: [u8; 32],
+			asset: [u8; 32],
+			// NOTE: Solidity balance is `U256` per default
+			amount: sp_core::U256,
```

Comment:
But we're using `u128` in other places as `TransferAssets`. Should this be unified?

Reply:
I had the same question ([-> slack thread](https://kflabs.slack.com/archives/C06FRHCRQL9/p1721981066605389?thread_ts=1721940738.924059&cid=C06FRHCRQL9)). The summary would be that this message initates a recovery of an arbitrary token (e.g. erc20 USDC). Amounts in EVM are u256 per default such that it makes sense to have the amount type in u256. Unfortunately, I had to fallback converting to `[u8; 32]` in the end because of serialization issues.

Reply:
Regarding the serialization, maybe you can a dd a wrapper type:

// Pseudocode
```rust
struct U256Wrapper(U256);
impl Serialize for U256Wrapper {
    fn serialize() {
        self.low_128().serialize()
        self.high_128().serialize()
    }
}
```

Also, maybe the `U256` already implements Serialize/Deserialize under some `serde` feature that we need to set (?)

Reply:
Maybe you need to use directly the dependency `primitive-types` enabling `serde`: https://docs.rs/primitive-types/0.12.2/primitive_types/

Reply:
Just some ideas, feel free to skip them

Reply:
@wischli, just to know if you read this

Reply:
Yeah I have read it but I didnt have the time yet to try it out, unblocking dev and demo is higher priority.

Reply:
As discussed on Slack, unfortunately the serialization is blocked by us missing serialization from `str` and `U256` being serialzed to `str` hex unfortunately. So will keep as is!